src/HtmlOptions/MatchRegularExpressionHtmlOptions.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Form\HtmlOptions;
+
+use Yiisoft\Validator\Rule\MatchRegularExpression;
+
+final class MatchRegularExpressionHtmlOptions implements HtmlOptionsProvider
+{
+    private MatchRegularExpression $validator;
+
+    public function __construct(MatchRegularExpression $validator)
+    {
+        $this->validator = $validator;
+    }
+
+    public function getHtmlOptions(): array
+    {
+        $options = $this->validator->getOptions();
+        return [
+            'pattern' => $options['pattern'],
```

Comment:
The input pattern is PCRE that isn't suitable for HTML. Should be passed through `Html::normalizeRegexpPattern()`.